cmd/cstor-pool-mgmt/controller/replica-controller/handler.go

```diff
@@ -522,3 +544,101 @@ func (c *CStorVolumeReplicaController) syncCvr(cvr *apis.CStorVolumeReplica) {
 		cvr.Status.Capacity = *capacity
 	}
 }
+
+func (c *CStorVolumeReplicaController) upgrade(cvr *apis.CStorVolumeReplica) (
+	*apis.CStorVolumeReplica, error,
+) {
+	var err error
+	if cvr.VersionDetails.Current != cvr.VersionDetails.Desired {
+		if !isCurrentVersionValid(cvr) {
+			return nil, pkg_errors.Errorf("invalid current version %s", cvr.VersionDetails.Current)
+		}
+		if !isDesiredVersionValid(cvr) {
+			return nil, pkg_errors.Errorf("invalid desired version %s", cvr.VersionDetails.Desired)
+		}
+		path := strings.Split(cvr.VersionDetails.Current, "-")[0] + "-" +
+			strings.Split(cvr.VersionDetails.Desired, "-")[0]
+		u := &upgradeParams{
+			cvr:    cvr,
+			client: c.clientset,
+		}
+		cvr, err = upgradeMap[path](u)
+		if err != nil {
+			return nil, err
+		}
+		cvr.VersionDetails.Current = cvr.VersionDetails.Desired
+		cvr, err = c.clientset.OpenebsV1alpha1().
+			CStorVolumeReplicas(cvr.Namespace).Update(cvr)
+		if err != nil {
+			return nil, pkg_errors.Wrap(
+				err,
+				"failed to update CVR with versionDetails",
```

Comment:
not required to return customize error, just retrun err

Reply:
Done